server/cli/init/base-file-sources/world.js

```diff
@@ -23,20 +24,46 @@ module.exports = function () {
     };
 
     /* eslint-disable new-cap */
-    this.Before(function (callback) {
+    this.Before(function (scenario, callback) {
     /* eslint-enable new-cap */
         global.httpBackend = new HttpBackend(global.browser);
         callback();
     });
 
     /* eslint-disable new-cap */
-    this.After(function (callback) {
+    this.After(function (scenario, callback) {
+        var browserlogArray = [];
+        var message;
     /* eslint-enable new-cap */
         global.httpBackend.clear();
-        global.browser.manage.deleteAllCookies();
+        global.browser.manage().deleteAllCookies();
         global.browser.executeScript('window.sessionStorage.clear();');
         global.browser.executeScript('window.localStorage.clear();');
-        callback();
+        if (scenario.isFailed()) {
+            global.browser.takeScreenshot().then(function (base64png) {
+                var decodedImage = new Buffer(base64png, 'base64').toString('binary');
+                scenario.attach(decodedImage, 'image/png', callback);
+            }, function (err) {

```

Comment:
I wouldn't usually recommend handling the error like this, and in fact the whole way promises are handled here is a bit weird.

I'd suggest having two functions `takeScreenshot` and `printBrowserLog` which both do their respective operations and return the promises, and then do something like `Promise.all([takeScreenshot(scenario), printBrowserLog()).catch(function (err) { callback(err) });`